nixos/modules/system/boot/zram-as-tmp.nix

```
@@ -0,0 +1,96 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.boot.tmp;
+in
+{
+  options = {
+    boot.tmp = with lib; {
+      useZram = mkOption {
+        type = types.bool;
+        default = false;
+        example = true;
+        description = ''
+           Whether to mount a zram device on {file}`/tmp` during boot.
+
+           ::: {.note}
+           Large Nix builds can fail if the mounted zram device is not large enough.
+           In such a case either increase the zramSettings.zram-size or disable this option.
+           :::
+        '';
+      };
+
+      zramSettings = {
+        zram-size = mkOption {
+          type = types.str;
+          default = "ram * 0.5";
+          example = "min(ram / 2, 4096)";
+          description = ''
+            The size of the zram device, as a function of MemTotal, both in MB.
+            For example, if the machine has 1 GiB, and zram-size=ram/4,
+            then the zram device will have 256 MiB.
+            Fractions in the range 0.1–0.5 are recommended
+
+            See: https://github.com/systemd/zram-generator/blob/main/zram-generator.conf.example
+          '';
+        };
+
+        compression-algorithm = mkOption {
+          type = types.str;
+          default = "zstd";
+          example = "lzo-rle";
+          description = ''
+            The compression algorithm to use for the zram device,
+            or leave unspecified to keep the kernel default.
```

```suggestion
            or leave unspecified to keep the kernel default.
```
The default is actually specified by us (not the kernel) as zstd.